File: JAVA/program/Utilities.java

import java.util.Vector;

public class Utilities {
    private Vector<Mahasiswa> listMhs = new Vector<Mahasiswa>();// <-- vector of Mahasiswa instance


    // contructor definition
    public Utilities(Vector<Mahasiswa> source) {
        this.listMhs = source;
    }

    // diplay in list format
    public void displayData(String[] label) {
        int index = 1;
        
        for (Mahasiswa n : this.listMhs) {
            System.out.println("=== Data Mhs: " + Integer.toString(index) + " ===");
            int index2 = 0;
            for (String att : n.getAllAttribute()) {

                System.out.println(label[index2] + ": " + att);
                index2++;
            }
            index++;
        }
    }
}

File: CPP/program/Utilities.cpp
#include "Utilities.hpp"

// contructor definition
Utilities::Utilities(vector<Mahasiswa> source)
{
    this->listMhs = source;
}

// display in list format
void Utilities::displayData(string label[])
{
    int index = 1;

    for (auto &it : listMhs)
    {
        cout << "=== Data Mhs: " + to_string(index) + " ===\n";
        int index2 = 0;
        cout << label[index2++] + ": " + it.getNik() << "\n";
        cout << label[index2++] + ": " + it.getNama() << "\n";
        cout << label[index2++] + ": " + it.getJenisKelamin() << "\n";
        cout << label[index2++] + ": " + it.getFakultas() << "\n";
        cout << label[index2++] + ": " + it.getProdi() << "\n";
        cout << label[index2++] + ": " + it.getAsalUniversitas() << "\n";
        cout << label[index2++] + ": " + it.getEmailEdu() << "\n";
        cout << label[index2++] + ": " + it.getNim() << "\n";
        index++;
    }
}

Utilities::~Utilities()
{
}